python_testspace_xml/testspace_xml.py

```diff
@@ -33,107 +33,57 @@ def __init__(self, name, comment):
-            with open(self.filePath, 'rb') as inFile:
-                out_file = tempfile.TemporaryFile()
-                gzip_file_path = out_file.name
-                out_file.close()
-                out_file = gzip.open(gzip_file_path, 'wb')
-                out_file.writelines(inFile)
-                out_file.close()
-                inFile.close()
-
-                # base64 the file
-                out_file = open(gzip_file_path, 'rb')
-                gzip_data = out_file.read()
-                b64_data = base64.standard_b64encode(gzip_data)
-                out_file.close()
-                os.remove(gzip_file_path)
-                cdata = dom.createCDATASection(b64_data)
-                anno.appendChild(cdata)
-
-        # add comments
-        for c in self.comments:
-            c_elem = dom.createElement('comment')
-            c_elem.setAttribute("label", c.name)
-            cdata = dom.createCDATASection(c.comment)
-            c_elem.appendChild(cdata)
-            anno.appendChild(c_elem)
-
-        parent_element.appendChild(anno)
-
-
-class TextAnnotation(Annotation):
-    def __init__(self, name, level='info', description=''):
-        Annotation.__init__(self, level, description)
-        self.name = name
-
-    def _write_xml(self, parent_element, dom):
-        # write as a file annotation to the root suite
-        anno = dom.createElement("annotation")
-        anno.setAttribute("description", XmlWriter.fix_line_ends(self.description))
-        anno.setAttribute("level", self.level)
-        anno.setAttribute("name", self.name)
+    def write_xml(self, parent_element, dom):
+        annotation = dom.createElement("annotation")
+        annotation.setAttribute("description", self.description)
+        annotation.setAttribute("level", self.level)
+        annotation.setAttribute("name", self.name)
+
+        if self.file_path is not None:
+            if not os.path.isfile(self.file_path):
+                # write as a warning text annotation
+                ta = Annotation(self.file_path, level='error')
+                ta.description = 'File: ' + self.file_path + ' not found.'
+                for com in self.comments:
+                    ta.comments.append(com)
+                ta.write_xml(parent_element, dom)
+                return
+            else:
+                annotation.setAttribute("link_file", "false")
+                annotation.setAttribute("file", "file://" + self.file_path)
+                annotation.setAttribute("mime_type", self.mimeType)
+                with io.open(self.file_path, 'rb') as inFile:
+                    out = BytesIO()
+                    with gzip.GzipFile(fileobj=out, mode="wb") as f:
+                        f.writelines(inFile)
+                    f.close()
+                    gzip_data = out.getvalue()
+                    b64_data = base64.standard_b64encode(gzip_data)
+                    cdata = dom.createCDATASection(b64_data.decode())
```

Comment:
The [decode ](https://docs.python.org/3.3/library/stdtypes.html#bytes.decode)is to convert b64_data from bytes to string.